app/test_main.py

```diff
@@ -1,3 +1,50 @@
+import pytest
+
 from app.main import get_human_age
 
-# write your code here
+
+class TestHumanCatDogYears:
+    @pytest.mark.parametrize(
+        "cat_years,dog_years,human_years",
+        [
+            (14, 14.22, [0, 0]),
+            (1, 1, [0, 0]),
+            (0, 0, [0, 0]),
+            (-3, -12, [0, 0]),
+            (15, 15, [1, 1]),
+            (23.9, 23.9, [1, 1]),
+            (24.0, 24.96, [2, 2]),
+            (27.96, 28.0, [2, 2]),
+            (28.0, 29.0, [3, 3]),
+            (10000, 10000, [2496, 1997]),
+        ],
+        ids=[
+            "`human_years` should equal `catdog_till_15` max",
+            "`human_years` should equal `catdog_till_15` min",
+            "should correctly work with zeros",
+            "should correctly work with negative numbers",
+            "`human_years` should equal `catdog_15_24` min",
+            "`human_years` should equal `catdog_15_24` max",
+            "`human_years` should equal `catdog_24` min",
+            "`human_years` should equal `catdog_24` max",
+            "`human_years` should equal `catdog_24+` min",
+            "`human_years` should equal `catdog_24+` big_num"
+        ]
+    )
+    def test_results_should_equal_correct_period(
+            self,
+            cat_years: int | float,
+            dog_years: int | float,
+            human_years: int | float
+    ) -> None:
+        assert get_human_age(cat_years, dog_years) == human_years
+
+
+class TestErrorTypes:
+    def test_type_error_missing_1_positional_argument(self) -> None:
+        with pytest.raises(TypeError, match=r".* missing 1 *."):
+            get_human_age(1)
```

Comment:
Interesting, but I think it's a redundant test because you will get the error in any case when there is a missing required argument, no need to check this case)

Reply:
I agree, but all those tests are redundant since we already got type annotations for arguments in the defined `app.main.get_human_age()`
So If someone who uses code provides data not according to sample, it's his logical error, not program...

Still, according to checklist, I should provide :
`if the function receives an incorrect type of data it raises the correct exception.`
which, for me, is the same redundant test, so I added some other of same kind, just in case

That's why I will change code only if checklist will be changed, your suggestions seems redundant to me)

Reply:
Tests are such a thing that they can be written endlessly if we are guided by the logic that we need to test every case. But there should be a reasonable limit to which cases you should test. If the QA wants to annoy the developer, he will make a lot of tests such like this ))))